clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/evaluation/EvaluationManager.kt

```diff
@@ -0,0 +1,503 @@
+            val jsonObject = whenTriggers[it] as? JSONObject
+            jsonObject?.let { nonNullJsonObject -> TriggerAdapter(nonNullJsonObject) }
+        }
+    }
+
+    internal fun getWhenLimits(limitJSON: JSONObject): List<LimitAdapter> {
+        val frequencyLimits = limitJSON.optJSONArray(Constants.INAPP_FC_LIMITS).orEmptyArray()
+        val occurrenceLimits = limitJSON.optJSONArray(Constants.INAPP_OCCURRENCE_LIMITS).orEmptyArray()
+
+        return (frequencyLimits.toList<JSONObject>() + occurrenceLimits.toList()).mapNotNull {
+            if (it.isNotNullAndEmpty()) {
+                LimitAdapter(it)
+            } else null
+        }.toMutableList()
+    }
+
+    /**
+     * Sorts list of InApp objects with priority(100 highest - 1 lowest) and if equal priority
+     * then then the one created earliest
+     */
+    internal fun sortByPriority(inApps: List<JSONObject>): List<JSONObject> {
+        val priority: (JSONObject) -> Int = { inApp ->
+            inApp.optInt(Constants.INAPP_PRIORITY, 1)
+        }
+
+        val ti: (JSONObject) -> String = { inApp ->
+            inApp.optString(Constants.INAPP_ID_IN_PAYLOAD, (Clock.SYSTEM.newDate().time / 1000).toString())
+        }
+        // Sort by priority descending and then by timestamp ascending
+        return inApps.sortedWith(compareByDescending<JSONObject> { priority(it) }.thenBy { ti(it) })
+    }
+
+    private fun shouldSuppress(inApp: JSONObject): Boolean {
+        return inApp.optBoolean(Constants.INAPP_SUPPRESSED)
+    }
+
+    @VisibleForTesting
+    internal fun suppress(inApp: JSONObject) {
+        val campaignId = inApp.optString(Constants.INAPP_ID_IN_PAYLOAD)
+        val wzrkId = generateWzrkId(campaignId)
+        val wzrkPivot = inApp.optString(Constants.INAPP_WZRK_PIVOT, "wzrk_default")
+        val wzrkCgId = inApp.optInt(Constants.INAPP_WZRK_CGID)
+
+        suppressedClientSideInApps.add(
+            mapOf(
+                Constants.NOTIFICATION_ID_TAG to wzrkId,
+                Constants.INAPP_WZRK_PIVOT to wzrkPivot,
+                Constants.INAPP_WZRK_CGID to wzrkCgId
+            )
+        )
+    }
+
+    /**
+     * Generates a unique WizRocket (Wzrk) identifier based on the provided input and current date.
+     *
+     * This method concatenates the provided identifier (`ti`) with the current date in the "yyyyMMdd" format
+     * to create a distinctive identifier for use in in-app notifications and tracking.
+     *
+     * @param ti The input identifier, often representing a campaign ID or similar.
+     * @param clock The Clock instance used for obtaining the current date. Defaults to the system clock.
+     *
+     * @return A unique WizRocket identifier combining the input identifier and the current date.
+     */
+    @VisibleForTesting
+    internal fun generateWzrkId(ti: String, clock: Clock = Clock.SYSTEM): String {
+        val dateFormatter = SimpleDateFormat("yyyyMMdd", Locale.getDefault())
```

Comment:
we can create `SimpleDateFormat` instance only once.

Reply:
added changes in https://github.com/CleverTap/clevertap-android-sdk/commit/7759328fda4527038de29e94693c661ce60356d0